src/gate/parser/bench/parser.cpp

```diff
@@ -0,0 +1,290 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the Utopia EDA Project, under the Apache License v2.0
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 ISP RAS (http://www.ispras.ru)
+//
+//===----------------------------------------------------------------------===//
+
+#include <FlexLexer.h>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <stdlib.h>
+#include <vector>
+
+#include "gate/model/gate.h"
+#include "gate/model/gnet.h"
+#include "gate/model/gsymbol.h"
+#include "headerFile"
+#include "tokens.h"
+
+extern "C" int scan_token();
+
+///Defines the compare method for TokenMap.name.
+#define COMPARE(text, line) do {\
+  if (yytext == i.name) {\
+    std::cout << "ERROR IN " << static_cast<const char*>(text) << " WITH "\
+      << yytext << "\tline: " << line << std::endl;\
+    exit(EXIT_FAILURE);\
+  }\
+} while (false)
+
+///Defines the alert method for debugging information.
+#define ALERT(errorString, line) do {\
+  std::cout << " expected '" << errorString << "'\tline: "\
+    << line << "\tСaught " << yytext << std::endl;\
+  exit(EXIT_FAILURE);\
+} while (false) 
+
+///Defines the method for verifying the next token.
+#define ASSERT_NEXT_TOKEN(expectedToken, errorString, place, line) do {\
+  Tokens token = getNextToken(place, line);\
+  if ( token != expectedToken ) {\
+    ALERT(errorString, line);\
+  }\
+} while (false)
+
+///Defines the adding in a token map vector.
+#define MAPS(definite, typeInit, maps) do {\
+  char* text = new char[strlen(yytext)];\
+  strncpy(text, yytext, strlen(yytext) + 1);\
+  TokenMap map {text, definite, typeInit};\
```

Comment:
- not enough memory allocated: C-strings are zero-terminated and requires an additional character
- manual allocation does not make sense after the move to `std::string`
```suggestion
  TokenMap map {yytext, definite, typeInit};\
```